server/routes.go

```diff
@@ -977,6 +988,10 @@ func streamResponse(c *gin.Context, ch chan any) {
 			return false
 		}
 
+		if err, ok := val.(error); ok {
+			val = gin.H{"error": err.Error()}
```

Comment:
it's much harder to inspect an error if it's wrapped in `gin.H` since it stores the string representation `error.Error()`. since some errors are sometimes `fmt.Errorf` with variable content, it's impossible to determine what the real error is without string matching, which is tedious and error prone

inspect the real error is necessary for stream=false but since `streamResponse` and `waitForStream` consumes the same channel, this needs to receive real errors